src/prof_recent.c

```diff
@@ -495,7 +546,47 @@ prof_recent_alloc_dump_node(emitter_t *emitter, prof_recent_t *n) {
 	emitter_json_object_end(emitter);
 }
 
-#define PROF_RECENT_PRINT_BUFSIZE 4096
+static void
+prof_recent_alloc_merge_locked(tsd_t *tsd, prof_recent_t *last) {
```

Comment:
nit: better to make the swap and merge utility not assuming the lists internally, but rather pass them in. That way it would be possible to test them individually.